sns/src/main/scala/akka/stream/alpakka/sns/SnsPublishSinkStage.scala

```diff
@@ -0,0 +1,77 @@
+package akka.stream.alpakka.sns
+
+import akka.Done
+import akka.stream.stage._
+import akka.stream.{Attributes, Inlet, SinkShape}
+import com.amazonaws.handlers.AsyncHandler
+import com.amazonaws.services.sns.AmazonSNSAsync
+import com.amazonaws.services.sns.model.{PublishRequest, PublishResult}
+
+import scala.concurrent.{Future, Promise}
+
+final class SnsPublishSinkStage(topicArn: String, snsClient: AmazonSNSAsync)
```

Comment:
An additional thought: It could be better to make this a flow, emitting the values downstream as they have been ack:ed from AWS, the factory for the simple use case as sink could then be achieved through composition (`SNSPublishFlow.to(Sink.ignore)`).